tests/sentry/api/bases/test_organization.py

```diff
@@ -310,6 +311,80 @@ def test_all_accessible_sigil_value_allow_joinleave(self):
         self.create_team_membership(user=self.user, team=self.team_1)
         self.run_test([self.project_1, self.project_2], project_ids=[-1])
 
+    @mock.patch("sentry.api.bases.organization.OrganizationEndpoint._get_projects_by_id")
```

Comment:
Just personal opinion but I feel like mocking methods like this is more likely to break in the future if the signature changes.

Reply:
Heh. i'm on the opposite side here.

IMO we should be decoupling our test dependencies and properly testing individual components.
We can still have integration tests, but if i'm trying to test a new feature, i shouldn't have to initialize an entire suite of unrelated dependencies.

IMO this is probably what's causing a lot of our flakes and making those flakes incredibly difficult to track down 🤷

Reply:
Yeah, I know the reasoning for doing unit tests like this, just my personal opinion that it produces tests that catch fewer issues in practice. 

fwiw our flakes are usually caused by date related issues in tests that are explicitly making queries. I don't think a lack of mocking is usually the cause here.